.github/actions/javascript/authorChecklist/newComponentCategory.ts

```diff
@@ -0,0 +1,104 @@
+import {parse} from '@babel/parser';
+import traverse from '@babel/traverse';
+import github from '@actions/github';
+import CONST from '../../../libs/CONST';
+import GithubUtils from '../../../libs/GithubUtils';
+
+type SuperClassType = {superClass: {name?: string; object: {name: string}; property: {name: string}} | null; name: string};
+
+const items = [
+    "I verified that similar component doesn't exist in the codebase",
+    'I verified that all props are defined accurately and each prop has a `/** comment above it */`',
+    'I verified that each file is named correctly',
+    'I verified that each component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone',
+    'I verified that the only data being stored in component state is data necessary for rendering and nothing else',
+    "In component if we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes",
+    'For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)',
+    'I verified that component internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)',
+    'I verified that all JSX used for rendering exists in the render method',
+    'I verified that each component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions',
+];
+
+function detectReactComponent(code: string, filename: string): boolean | undefined {
+    if (!code) {
+        console.error('failed to get code from a filename', code, filename);
+        return;
+    }
+    const ast = parse(code, {
+        sourceType: 'module',
+        plugins: ['jsx'], // enable jsx plugin
```

Comment:
we need to include the `typescript` plugin as well:

```suggestion
        plugins: ['jsx', 'typescript'],
```

Reply:
Good catch 👍 I will also add tests with TS components to make sure it works fine